### PR TITLE
Rds/kjw/rule 23 16

### PR DESCRIPTION
**Pull request by KarenWGard**

### CODE REVIEW

File: docs/section23/Rule23-16.md [hunk 1]
```diff
@@ -5,42 +5,43 @@
...
+**Rule Assertion:** B-RMR = expected value  
 **Appendix G Section:** Section 23 Air-side  
 **90.1 Section Reference:** Section G3.1.3.19 Preheat Coils (Systems 5 through 8)  
 **Data Lookup:** None  
 **Evaluation Context:** Building  
 
 **Applicability Checks:**  
 
-1. B-RMI is modeled with at least one air-side system that is Type-5, 6, 7, 8, 7a, 8a, 5b, 6b, 7b, 8b, 7c.  
+1. B-RMR is modeled with at least one air-side system that is Type-5, 6, 7, 8, 7a, 8a, 5b, 6b, 7b, 8b, 7c.
```
Comment (mkarpman) [inline docs/section23/Rule23-16.md:16]: Line 56 expects preheat_b.heating_system_type == "FLUID_LOOP" in order for the True outcome, but shouldn'd preheat be electric resistance for Systems 6 and 8? Recommend removing this condition.

File: docs/section23/Rule23-16.md [hunk 2]
```diff
@@ -5,42 +5,43 @@
...
-- For each baseline system type in the baseline_hvac_system_dict: `for baseline_system_type in baseline_hvac_system_dict:`
-  - check if the baseline_system_type is one of 6, 8, 8a, 6b, 8b: `if baseline_system_type in ["SYS-5", "SYS-6", "SYS-7", "SYS-8", "SYS-7A", "SYS-8A", "SYS-5B", "SYS-6B", "SYS-7B", "SYS-8B", "SYS-7C"]:`
-    - add the hvac_ids for this system type to the list of eligible_systems: `eligible_systems += baseline_hvac_system_dict[baseline_system_type]`
 
-- For each zone in B_RMI: `for zone_b in B_RMI...zones:`
+- make a list of eligible hvac systems: `eligible_systems = []`
+- loop through the system types in baseline_hvac_system_dict: `for system_type in baseline_hvac_system_dict:`
+  - check if it is one of the target baseline system types: `if any(baseline_system_type_compare(system_type,target_system_type,false) for target_system_type in target_system_types):`
+    - add the hvac system ids to the list of eligible systems: `eligible_systems = eligible_systems + baseline_hvac_system_dict[system_type]`
+- For each zone in B_RMR: `for zone_b in B_RMR...zones:`
```
Comment (mkarpman) [inline docs/section23/Rule23-16.md:38]: Why not use the existing function get_dict_of_zones_and_terminal_units_served_by_hvac_sys instead to get zones that are served by each eligible system and then follow logic in lines 46 and 48 for these zones?

File: docs/section23/Rule23-16.md (hunk 1 above)
Comment (KarenWGard) [reply docs/section23/Rule23-16.md:16]: the work that I've done on this rule is only related to schema updates, but this makes sense.  Removed both references to water coils:

`( preheat_b.heating_system_type == "FLUID_LOOP" ) AND ( get_component_by_id(preheat_b.hot_water_loop).type == "HEATING" )`

The only check now is related to the preheat coil setpoint temperature: 

`( preheat_b.heating_coil_setpoint == hvac_max_zone_setpoint_dict[hvac_b] - 20F )`

File: docs/section23/Rule23-16.md (hunk 2 above)
Comment (KarenWGard) [reply docs/section23/Rule23-16.md:38]: updated

**Pull request merged**